persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/connectionfactory/PooledConnectionFactory.java

```diff
@@ -14,86 +25,162 @@
 import org.infinispan.persistence.spi.PersistenceException;
 import org.infinispan.util.logging.LogFactory;
 
-import com.mchange.v2.c3p0.ComboPooledDataSource;
-import com.mchange.v2.c3p0.DataSources;
-
 /**
- * Pooled connection factory based on C3P0. For a complete configuration reference, look <a
- * href="http://www.mchange.com/projects/c3p0/index.html#configuration">here</a>. The connection pool can be configured
- * in various ways, as described <a href="http://www.mchange.com/projects/c3p0/index.html#configuration_files">here</a>.
- * The simplest way is by having an <tt>c3p0.properties</tt> file in the classpath. If no such file is found, default,
- * hardcoded values will be used.
+ * Pooled connection factory that uses HikariCP by default. If no Hikari system properties or property files exist on the
+ * classpath and a C3P0 properties file is found then we utilise the C3P0 connection pool.  Note, that support for the
+ * C3PO pool has been deprecated and will be removed in future versions.
+ *
+ * HikariCP property files can be specified by either passing the <tt>hikaricp.configurationFile</tt> system property with
+ * the path to a properties file, or by ensuring that a <tt>hikari.properties</tt> file is on the classpath. Note, that
+ * the <tt>hikaricp.configurationFile</tt> takes precedence over <tt>hikari.properties</tt>.
+ *
+ * For a complete configuration reference for C3P0 look <a href="http://www.mchange.com/projects/c3p0/index.html#configuration">here</a>.
+ * The connection pool can be configured n various ways, as described
+ * <a href="http://www.mchange.com/projects/c3p0/index.html#configuration_files">here</a>. The simplest way is by having
+ * an <tt>c3p0.properties</tt> file in the classpath.
+ *
+ * If no properties files are found for either HikariCP or C3PO then the default values of these connection pools are
+ * utilised.
  *
  * @author Mircea.Markus@jboss.com
  * @author Tristan Tarrant
+ * @author Ryan Emerson
  */
 public class PooledConnectionFactory extends ConnectionFactory {
 
-   private static final Log log = LogFactory.getLog(PooledConnectionFactory.class, Log.class);
-   private static final boolean trace = log.isTraceEnabled();
-   private ComboPooledDataSource pooledDataSource;
+   private static final Log LOG = LogFactory.getLog(PooledConnectionFactory.class, Log.class);
+   private static final boolean TRACE = LOG.isTraceEnabled();
+   private static final String FORCE_C3P0 = "c3p0.force";
```

Comment:
@tristantarrant The reason I didn't drop c3p0 completely, is that by moving from c3p0 to Hikari, we are essentially changing the api because the two connection pools utilise different properties files. So if removed c3p0 without deprecation a user migrating from 8.2.x to 9.0 would need to transition to hikari.properties without any prior warning. Of course, if a user does not utilise a .properties file to provided advanced configuration this is a non issue.